car_directory/dir.go

```diff
@@ -53,16 +53,25 @@ func (c *Car) Render(out chan<- string) {
 	// Calculate max directory elements to display.
 	maxLength := 3
 	if e := os.Getenv("BULLETTRAIN_CAR_DIRECTORY_MAX_LENGHT"); e != "" {
-		if ml, err := strconv.Atoi(e); err == nil && ml >= 3 {
+		if ml, err := strconv.Atoi(e); err == nil && ml >= 0 {
 			maxLength = ml
 		}
 	}
 
 	// Compose directory segments.
 	ps := string(os.PathSeparator)
 	dirs := strings.Split(dir, ps)
-	if maxLength > 0 && len(dirs) > maxLength+1 {
-		newPath := make([]string, 0)
+	newPath := make([]string, 0)
+
+	if maxLength == 1 {
+		if len(dirs) == 2 {
```

Comment:
It does not really matter how long the path is if you'd only want to get the last segment. Simplify it by just returning the last element of `dir`.

Reply:
Note: do not use `newPath` here as it is not even needed here.

Reply:
It does matter. If the pwd is `/` only, then `len(dirs) == 2` . And this is exactly what this condition tests. If I return just last bit of the path then in the case of `/`, there is nothing displayed.